### PR TITLE
Create and edit vendor unit test

### PR DESCRIPTION
**Pull request by RevathyRajeevan**
Added unit testing for vendor creation and editing
image

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -1,2 +1,4 @@
 <!-- <app-nx-welcome></app-nx-welcome> <router-outlet></router-outlet> -->
-<app-vendor-listing></app-vendor-listing>
\ No newline at end of file
+<!-- <app-vendor-listing></app-vendor-listing> -->
```
Comment (kavyamolr) [inline src/app/app.component.html:2]: Remove commented code

File: src/app/app.component.spec.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import { TestBed } from '@angular/core/testing';
 import { AppComponent } from './app.component';
-import { NxWelcomeComponent } from './nx-welcome.component';
+//import { NxWelcomeComponent } from './nx-welcome.component';
```
Comment (kavyamolr) [inline src/app/app.component.spec.ts:3]: Remove commented codes

File: src/app/app.module.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import {
+  BrowserModule,
+  provideClientHydration,
+} from '@angular/platform-browser';
+import { RouterModule } from '@angular/router';
+import { AppComponent } from './app.component';
+import { routes } from './app.routes';
+import { provideAnimations } from '@angular/platform-browser/animations';
+import { provideHttpClient } from '@angular/common/http';
+//import { NxWelcomeComponent } from './nx-welcome.component';
```
Comment (kavyamolr) [inline src/app/app.module.ts:11]: Remove

File: src/app/app.component.html (hunk 1 above)
Comment (RevathyRajeevan) [reply src/app/app.component.html:2]: done

File: src/app/app.component.spec.ts (hunk 2 above)
Comment (RevathyRajeevan) [reply src/app/app.component.spec.ts:3]: done

File: src/app/app.module.ts (hunk 3 above)
Comment (RevathyRajeevan) [reply src/app/app.module.ts:11]: done

File: src/app/features/vendor/vendor-creation/vendor-creation.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+  it('should add the vendor data during  form submission', () => {
+    const { component, spyObj } = setup();
+    component.isEdit = false;
+    component.addVendorForm = getVendorServiceFormMock();
+    component.submitVendor();
+    expect(spyObj.addVendorSpy).toHaveBeenCalled();
+  });
+
+  it('should handle validation error during vendor form submission', () => {
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.spec.ts:228]: Description should not be technical. 
eg: should show error message when there is an error occurs during vendor creation / update

**Pull request closed without merging**